src/pytest/test_cursors.py

```diff
@@ -102,10 +102,16 @@ def testTimeout(env):
     loadDocs(env, idx='idx1')
     # Maximum idle of 1ms
     q1 = ['FT.AGGREGATE', 'idx1', '*', 'LOAD', '1', '@f1', 'WITHCURSOR', 'COUNT', 10, 'MAXIDLE', 1]
-    resp = env.cmd( * q1)
-    sleep(0.01)
-    env.cmd('FT.CURSOR', 'GC', 'idx1', '0')
-    env.assertEqual(0, getCursorStats(env, 'idx1')['index_total'])
+    resp = env.cmd(*q1)
+    exptime = time() + 1.5
+    rv = 1
+    while time() < exptime:
+        sleep(0.01)
+        env.cmd('FT.CURSOR', 'GC', 'idx1', '0')
```

Comment:
@mnunberg shouldn't it be cleaned after the first GC call?

Reply:
It sometimes isn't, depending on how accurate the system clocks are in Python. I'm just allowing a margin of error of 1.5s, instead of sleeping.